src/lib/PageTracker.js

```diff
@@ -0,0 +1,382 @@
+    idleTimerTimeoutId;
+
+    /** @property {boolean} - Whether the Access Stats Preview Event was reported */
+    previewEventReported = false;
+
+    /** @property {string|null} - File Owner Enterprise ID. */
+    ownerEId;
+
+    /** @property {number} - Interval ID for reporting page data */
+    reportingIntervalId;
+
+    /** @property {string|null} - Random ID to identify a particular Session/Visit */
+    sessionId;
+
+    /** @property {Date|null} - Reference date from where the current report to be sent starts */
+    startTime;
+
+    /** @property {string|null} - User Enterprise ID */
+    userEId;
+
+    /** @property {boolean} - Whether the user events are binded or not */
+    userEventsBounded = false;
+
+    /** @property {number|null} - User ID */
+    userId;
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * [constructor]
+     * @param {Object} options - Content Insights Related Options
+     * @param {Object} file - File that is being previewed
+     * @return {PageTracker} PageTracker instance
+     */
+    constructor(options = {}, file) {
+        super();
+        // Generate a new session id. This would need to be later changed to be generated on server side
+        this.sessionId = uuidv4();
+        this.setOptions(options);
+        this.file = file;
+        this.handleUserActive = this.handleUserActive.bind(this);
+        this.handleUserInactive = this.handleUserInactive.bind(this);
+        this.throttledActiveUserHandler = throttle(this.handleUserActive, ACTIVE_USER_THROTTLE, { trailing: false });
+    }
+
+    /**
+     * [destructor]
+     *
+     * @return {void}
+     */
+    destroy() {
+        this.stopTracking();
+        this.unbindIdleUserEvents();
+        this.clearIdleTimeout();
+        this.currentPage = null;
+        this.file = {};
+        this.sessionId = null;
+        this.startTime = null;
+    }
+
+    /**
+     * Get the session id
+     *
+     * @public
+     * @return {string}
+     */
+    getSessionId() {
+        return this.sessionId;
+    }
+
+    /**
+     * Call the interval starting function and set the idle timer
+     *
+     * @public
+     * @return {void}
+     */
+    init() {
+        if (!this.userEventsBounded) {
+            this.bindIdleUserEvents();
+        }
+        // Start tracking time on file if content insights is turn on for that particular file
+        this.setIdleTimer();
+        this.startTracking();
+    }
+
+    /**
+     * Wheter Advanced Content Insights is on/off
+     *
+     * @public
+     * @return {boolean}
+     */
+    isActive() {
+        return this.isAdvancedInsightsActive;
+    }
+
+    /**
+     * Start the content insights interval, to report data each number of seconds defined on a Const
+     *
+     * @public
+     * @return {void}
+     */
+    startTracking() {
```

Comment:
Is there a max number of seconds we should track?

Reply:
There is no upper limit, but we stop emitting events after about 30 seconds (defined in this class as IDLE_TIMER). 